include/deal.II/grid/reference_cell.h

```diff
@@ -381,10 +358,35 @@ namespace ReferenceCells
   DEAL_II_CONSTEXPR const ReferenceCell Invalid =
     internal::ReferenceCell::make_reference_cell_from_int(
       static_cast<std::uint8_t>(-1));
+
+  /**
+   * Return the correct simplex reference cell type for the given dimension
+   * `dim`. Depending on the template argument `dim`, this function returns a
+   * reference to either Vertex, Triangle, or Tetrahedron.
+   */
+  template <int dim>
+  constexpr const ReferenceCell &
+  get_simplex();
+
+  /**
+   * Return the correct hypercube reference cell type for the given dimension
+   * `dim`. Depending on the template argument `dim`, this function returns a
+   * reference to either Vertex, Quadrilateral, or Hexahedron.
+   */
+  template <int dim>
+  constexpr const ReferenceCell &
+  get_hypercube();
 } // namespace ReferenceCells
 
 
 
+inline DEAL_II_CONSTEXPR
+ReferenceCell::ReferenceCell()
+  : ReferenceCell(ReferenceCells::Invalid)
+{}
```

Comment:
...to here, and also changed the initializer: It's now explicitly `Invalid`, rather than duplicating the place where we say that that corresponds to `-1`. This way, we keep to the way that the only places where one can generate specific objects is in namespace `ReferenceCells`.